lib/bank_account/checking_account.rb

```diff
@@ -0,0 +1,64 @@
+require "pry"
+
+module BankAccount
+
+  class CheckingAccount < Account
+
+    # This is unique to the checking account
+    attr_accessor :free_checks
+
+    def initialize(id, initial_balance)
+      super
+      @free_checks = 3 #Per month
+    end
+
+    # Get monies
+    def withdraw(amount)
+
+      slight_burglary = 1 #That yucky fee
+      total_withdraw = amount + slight_burglary
+
+      # Proceeds if you have enough monies
+      if @current_balance - total_withdraw > 0
+        @current_balance -= total_withdraw
+      else
+        # Yells at you (nicely) if you don't have enough monies
+        return "Sorry, you can't have a negative balance in your account! Withdraw canceled. Let's keep your current balance at #{@current_balance}."
+      end
+    end
+
+    def withdraw_using_check(amount)
+      highway_robbery = 2 #Yucky fee
+      total_withdraw = amount + highway_robbery
+
+      # Resetting @free_checks to 3 each subsequent month
+      # Grabs the calendar month when a check is first used for withdraw
+      first_time_use = Time.new.month
+      # Sets an "alarm" for next month
+      next_month = first_time_use + 1
+      # Runs a check
+      if Time.now.month == next_month
+        @free_checks = 3
+      end
+      # I don't think this works the way I want it to.
```

Comment:
I like that you attempted to use the time functionality to reset the check amounts! I think you would actually need some sort of ongoing "job" that would reset the checks every month. Another option would be keeping track of the # of days that have elapsed between the last time it was reset.